be/src/storage/column_expr_predicate.cpp

```diff
@@ -285,32 +285,62 @@ Status ColumnExprPredicate::try_to_rewrite_for_zone_map_filter(starrocks::Object
 }
 Status ColumnExprPredicate::seek_inverted_index(const std::string& column_name, InvertedIndexIterator* iterator,
                                                 roaring::Roaring* row_bitmap) const {
-    // Only support simple like predicate for now
-    // Root must be LIKE, and left child must be ColumnRef, which satisfy simple like predicate
-    // format as: col LIKE xxx, xxx can be any expr with string type
+    // Only support simple (NOT) LIKE/MATCH predicate for now
+    // Root must be (NOT) LIKE/MATCH, and left child must be ColumnRef, which satisfy simple (NOT) LIKE/MATCH predicate
+    // format as: col (NOT) LIKE/MATCH xxx, xxx can be any expr with string type
+    // For the untokenized mode, LIKE and MATCH are both available.
+    // Otherwise, only MATCH is available.
     bool vaild_pred = false;
-    auto* vectorized_function_call = dynamic_cast<VectorizedFunctionCallExpr*>(_expr_ctxs[0]->root());
-    if (vectorized_function_call &&
-        LIKE_FN_NAME == boost::to_lower_copy(vectorized_function_call->get_function_desc()->name) &&
-        _expr_ctxs[0]->root()->get_num_children() == 2 &&
-        _expr_ctxs[0]->root()->get_child(0)->node_type() == TExprNodeType::SLOT_REF) {
-        vaild_pred = true;
-    }
+    bool vaild_like = false;
+    bool vaild_match = false;
+    bool with_not = false;
+
+    with_not = _expr_ctxs[0]->root()->node_type() == TExprNodeType::COMPOUND_PRED &&
+               _expr_ctxs[0]->root()->op() == TExprOpcode::COMPOUND_NOT;
+    DCHECK((with_not && _expr_ctxs[0]->root()->get_num_children() == 1) || !with_not);
+
+    Expr* expr = with_not ? _expr_ctxs[0]->root()->get_child(0) : _expr_ctxs[0]->root();
+
+    auto* vectorized_function_call =
+            expr->node_type() == TExprNodeType::FUNCTION_CALL ? down_cast<VectorizedFunctionCallExpr*>(expr) : nullptr;
+
+    // check if satisfy vaild LIKE format
+    vaild_like = vectorized_function_call != nullptr &&
+                 LIKE_FN_NAME == boost::to_lower_copy(vectorized_function_call->get_function_desc()->name) &&
+                 expr->get_num_children() == 2 && expr->get_child(0)->node_type() == TExprNodeType::SLOT_REF;
+
+    // check if satisfy vaild MATCH format
+    vaild_match = vectorized_function_call == nullptr && expr->node_type() == TExprNodeType::MATCH_EXPR &&
```

Comment:
`match` is not function?

Reply:
I have defined it as MATCH_EXPR which is a new type